dotCMS/src/main/java/com/dotcms/publisher/util/DependencySet.java

```diff
@@ -108,7 +116,43 @@ private boolean addOrClean ( String assetId, Date assetModDate, Boolean cleanFor
             for (Environment env : envs) {
 				PushedAsset asset;
 				try {
-					asset = APILocator.getPushedAssetsAPI().getLastPushForAsset(assetId, env.getId());
+					if(endpointIds == null){
+						//this code allows to find if the asset should be included or not in the dependencies depending of the assetId, environment Id and endpoint Ids 
+						List<PublishingEndPoint> allEndpoints = APILocator.getPublisherEndPointAPI().findSendingEndPointsByEnvironment(env.getId());
+						List<String> endpoints = new ArrayList<String>();
+		                //Filter Endpoints list
+		                for(PublishingEndPoint ep : allEndpoints) {
+		                    if(isStatic && ep.isEnabled() && AWSS3Publisher.PROTOCOL_AWS_S3.equals(ep.getProtocol())) {
+		                    	//If the isStatic variable is true then get all the static endpoints
+		                    	endpoints.add(ep.getId());
```

Comment:
No, in this case, the line should remain there. We don't want to include all the endpoints, only the ones that are static endpoints if the variable isStatic is true or the dynamic endpoints if the isStatic variable is false. The validation is there because could exist environments with mixed endpoints (static and dynamic under the same environment)